backend/stakepoold/server.go

```diff
@@ -174,86 +219,156 @@ func runMain() int {
+	log.Infof("default voting config: VoteVersion %v VoteBits %v", votingConfig.VoteVersion,
 		votingConfig.VoteBits)
 
-	// TODO re-work main loop
-	// should be something like this:
-	// loadData()
-	// if ticket/user voting prefs -> enable voting -> refresh
-	// if no ticket/user voting prefs -> pull from db/wallets -> enable voting
-
 	var userdata = &userdata.UserData{}
 	userdata.DBSetConfig(cfg.DBUser, cfg.DBPassword, cfg.DBHost, cfg.DBPort, cfg.DBName)
 
 	userVotingConfig, err := userdata.MySQLFetchUserVotingConfig()
 	if err != nil {
-		log.Infof("could not obtain voting config: %v", err)
-		return 12 // wtf
+		log.Errorf("could not obtain voting config from MySQL: %v", err)
+	} else {
+		log.Infof("loaded prefs for %d users from MySQL", len(userVotingConfig))
+	}
+
+	err = txrules.IsValidPoolFeeRate(cfg.PoolFees)
+	if err != nil {
+		log.Errorf("poolfees is invalid: %v", err)
+		return err
 	}
 
 	ctx := &appContext{
-		blockheight:        0,
-		dataPath:           dataPath,
-		feeAddrs:           feeAddrs,
-		params:             activeNetParams.Params,
-		quit:               make(chan struct{}),
-		reloadUserConfig:   make(chan struct{}),
-		reloadTickets:      make(chan struct{}),
-		userData:           userdata,
-		userVotingConfig:   userVotingConfig,
-		votingConfig:       &votingConfig,
-		walletConnection:   walletConn,
-		winningTicketsChan: make(chan WinningTicketsForBlock),
-		testing:            false,
-	}
-
-	ctx.ticketsMSA, _ = walletFetchUserTickets(ctx)
+		dataPath:               cfg.DataDir,
+		feeAddrs:               feeAddrs,
+		feePercent:             cfg.PoolFees,
+		newTicketsChan:         make(chan NewTicketsForBlock),
+		params:                 activeNetParams.Params,
+		quit:                   make(chan struct{}),
+		reloadUserConfig:       make(chan struct{}),
+		reloadTickets:          make(chan struct{}),
+		spentmissedTicketsChan: make(chan SpentMissedTicketsForBlock),
+		userData:               userdata,
+		userVotingConfig:       userVotingConfig,
+		votingConfig:           &votingConfig,
+		walletConnection:       walletConn,
+		winningTicketsChan:     make(chan WinningTicketsForBlock),
+		testing:                false,
+	}
 
 	// Daemon client connection
 	nodeConn, nodeVer, err := connectNodeRPC(ctx, cfg)
 	if err != nil || nodeConn == nil {
 		log.Infof("Connection to dcrd failed: %v", err)
-		return 6
+		return err
 	}
 	ctx.nodeConnection = nodeConn
 
 	// Display connected network
 	curnet, err := nodeConn.GetCurrentNet()
 	if err != nil {
 		log.Errorf("Unable to get current network from dcrd: %v", err)
-		return 7
+		return err
 	}
 	log.Infof("Connected to dcrd (JSON-RPC API v%s) on %v",
 		nodeVer.String(), curnet.String())
 
-	_, height, err := nodeConn.GetBestBlock()
+	// prune save data
+	err = pruneData(ctx)
 	if err != nil {
-		log.Errorf("unable to get bestblock from dcrd: %v", err)
-		return 8
+		log.Warnf("pruneData error: %v", err)
+	}
+
+	// not used currently
```

Comment:
I made the mutex nameless.  As I said previously, it had problems with the gob encoder but I forgot I rewrote it a bit so it's no longer a problem.

I also cleaned up the comments.